File: atom-backend/atom-service/src/main/java/cn/iinti/atom/service/base/dbcache/exs/UserEx.java
package cn.iinti.atom.service.base.dbcache.exs;

import cn.iinti.atom.entity.UserInfo;
import cn.iinti.atom.service.base.perm.PermsService;

import java.util.Collection;
import java.util.Collections;
import java.util.Map;
import java.util.Objects;

public class UserEx {
    public Map<String, Collection<String>> perms = Collections.emptyMap();

    private String prePermsConfig = null;

    public void reload(UserInfo userInfo) {
        if (Objects.equals(prePermsConfig, userInfo.getPermission())) {
            return;
        }
        prePermsConfig = userInfo.getPermission();
        perms = PermsService.parseExp(prePermsConfig, true);
    }
}

File: server/src/main/kotlin/cn/iinti/atom/service/base/dbcache/exs/UserEx.kt
package cn.iinti.atom.service.base.dbcache.exs

import cn.iinti.atom.entity.UserInfo
import cn.iinti.atom.service.base.perm.PermsService

class UserEx {
    var perms: Map<String, Collection<String>> = emptyMap()
        private set

    private var prePermsConfig: String? = null

    fun reload(userInfo: UserInfo) {
        if (prePermsConfig == userInfo.permission) {
            return
        }
        prePermsConfig = userInfo.permission
        perms = PermsService.parseExp(prePermsConfig!!, true)
    }
}